### PR TITLE
Adds a new Hockey bloodsprts themed traitor item set.

### PR DESCRIPTION
**Pull request by GuyonBroadway**
Adds a potent hockey themed bundle of items for traitors, that includes a potent set of armour with helmet, a two-handed weapon mounted on a backpack that is the bastard child of a powerfist and baseball bat, some fancy pants looking roller-skates, A belt that can deploy a special thrown item that deals damage, stamina damage and stuns a target upon hitting them, this has a cooldown of about ten seconds and has slots to store up to two of them, and a hockey skull mask for badass points.

This set is purchasable from the traitor uplink for 20tc.

The whole set is NODROP, once you put this shit on you go all in, the idea is it's like a chainsaw level rampage item on steroids.

The armour is syringe proof, fireproof and spaceworthy if worn with the helmet and mask and they all have some decent energy resistance. However the potent hockey stick weapon can only be deployed from it's back mounted holder which cannot be removed, so in order to fully use it's power you need to sacrifice some versatility. Also if you get stunbaton'd you are down for 10 like anyone else.

The belt deploys little holopucks to your hand that deal about 10 brute and 20 stamina damage when thrown, this is amplified by the baseball bat effect from the hockey stick. They can be caught and thrown back at you so be careful.

The stick deals 18 damage and has a powerfist like knockback effect. Due to how the throw code works if they hit a wall they take 10 extra damage that pierces armour and stuns them for a little while. Leaving them helpless as you slap them into the wall over and over. It must be used with both hands to make use of it's damage but you need a free hand to throw (and bat back) objects.

Sprites are made by Fury McFlurry (except the puck sprite, that's just the middle of the E-sheild)

Possible changes that may or may not be necessary.

-Buff/Nerf the damage of the pucks and/or the stick itself.
-Buff/nerf the protective values of the armour, including its resistances to space/fire/syringes
-Possibly make an item that strips you down and equips the set directly on you so people can't cherry pick the gear.(This is a snowflakey solution)

hockeyset
The reason the right foot looks a little stretched is because the hockey stick is held down like a real hockey stick when you onehand it so it lines up with the skates.

### CODE REVIEW

**Comment by Hackole** [comment]
If it's all sorted out codewise +1, this is neat as holy hell

**Comment by GuyonBroadway** [comment]
It's all been tested out and works, the only issue was Icon conflicts last PR.

**Comment by ghost** [comment]
The prospect of thawcking someone across the station with a hockey stick is killer. +1

**Comment by FuryMcFlurry** [comment]
sprites were in the work in 2016, thats crazy long.
 Glad it looks great ingame.

**Comment by JohnGinnane** [comment]
Half way through code review, will continue tomorrow

**Comment by SpankMasterson** [comment]
+1 contingent on successful code review

**Comment by Swagile** [comment]
same, GM + 1 if code check passes

a brand spanking new feature is always welcome and we can revert it if its too OP

File: code/game/objects/items/weapons/hockey.dm [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+	target.visible_message("<span class='danger'>[target.name] was pucked by [user] 'eh!</span>", \
+		"<span class='userdanger'>You hear a loud crack 'eh!</span>", \
+		"<span class='italics'>You hear the sound of bones crunching 'eh!</span>")
+
+	var/atom/throw_target = get_edge_target_turf(target, get_dir(src, get_step_away(target, src)))
+	spawn(1)
+		target.throw_at(throw_target, 10, 0.2)	//Throws the target 10 tiles
+
+	playsound(loc, 'sound/weapons/resonator_blast.ogg', 50, 1)
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/hockey.dm]: Same but for specthrowsound

File: code/game/objects/items/weapons/hockey.dm [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+	throwforce = 30
+
+
+/mob/living/carbon/human/hitby(atom/movable/PP, blocked = 0)
+	..()
+	if(blocked <= 0)
+		if(istype(PP, /obj/item/holopuck))
+			apply_effect(2, STUN)
+			playsound(src, 'sound/effects/snap.ogg', 50, 1)
+			visible_message("<span class='danger'>[src] has been dazed by a holopuck!</span>", \
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/hockey.dm]: Don't use src, use name or something

**Review by JohnGinnane — CHANGES_REQUESTED** [review]

**Comment by GuyonBroadway** [comment]
Requested changes made.

File: code/datums/uplink_item.dm [hunk 3]
```diff
@@ -525,6 +525,13 @@ var/list/uplink_items = list()
 	excludefrom = list(/datum/game_mode/nuclear,/datum/game_mode/gang)
 	cost = 20
 
+/datum/uplink_item/dangerous/hockey
+	name = "Ka-Nada Boxed S.S.F Hockey Set"
+	desc = "The iconic extreme environment gear used by Ka-Nada special sport forces. Used to devastating effect during the great northern sports wars of the second great athletic republic. The unmistakable grey and red gear provides great protection from most if not all environmental hazards and combat threats in addition to coming with the signature weapon of the Ka-Nada SSF and all terrain Hyper-Blades for enhanced mobility and lethality in melee combat. This power comes at a cost as your Ka-Nada benefactors expect absolute devotion to the cause, once equipped you will be unable to remove the gear so be sure to make it count."
```
Comment (JohnGinnane) [inline code/datums/uplink_item.dm:530]: Okay so I have some questions on this:

Shouldn't "great northern sports wars" and "second great athletic republic" be capitalised?

Also lack of commas; "great protection from most, if not all, environmental hazards and combat threats, in addition to.... weapon of the Ka-Nada SSF, and all terrain..."

File: code/game/objects/items/weapons/hockey.dm [hunk 4]
```diff
@@ -0,0 +1,265 @@
+/obj/item/weapon/hockeypack
+	name = "Ka-Nada Special Sport Forces Hockey Pack"
+	desc = "Holds and powers a Ka-Nada SSF Hockey Stick, A devastating weapon capable of knocking men around like toys and batting objects at deadly velocities."
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/hockey.dm:3]: Uncapitalise "A devastating"

File: code/game/objects/items/weapons/hockey.dm [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+		return 1
+
+/obj/item/weapon/twohanded/hockeystick/Move()
+	..()
+	if(loc != pack.loc)
+		loc = pack.loc
+
+/obj/item/weapon/storage/belt/hockey
+	name = "Holopuck Generator"
+	desc = "A Belt mounted device that quickly fabricates hard-light holopucks that when thrown will stall and slow down foes dealing minor damage. Has a pouch to store a pair of spare pucks"
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/hockey.dm:195]: "holopucks that, when thrown, will stall.."

Comment (JohnGinnane) [reply code/game/objects/items/weapons/hockey.dm:195]: "slow down does, dealing minor damage."

File: code/modules/clothing/head/helmet.dm [hunk 6]
```diff
@@ -351,3 +351,16 @@
 		if(F.on)
 			user.AddLuminosity(-F.brightness_on)
 			SetLuminosity(F.brightness_on)
+
+/obj/item/clothing/head/helmet/hockey
+	name = "Ka-Nada winter sport combat helmet."
```
Comment (JohnGinnane) [inline code/modules/clothing/head/helmet.dm:356]: Is it "winter sport" or "winter sports"?

File: code/modules/clothing/masks/gasmask.dm [hunk 7]
```diff
@@ -192,3 +192,10 @@ obj/item/clothing/mask/gas/tiki_mask/attack_self(mob/user)
 		icon_state = options[choice]
 		M << "The Tiki Mask has now changed into the [choice] Mask!"
 	return 1
+
+/obj/item/clothing/mask/gas/hockey
+	name = "Ka-Nada Hokcey Mask"
```
Comment (JohnGinnane) [inline code/modules/clothing/masks/gasmask.dm:197]: "Hockey"

File: code/modules/clothing/suits/armor.dm [hunk 8]
```diff
@@ -416,3 +416,19 @@
 			M.adjustFireLoss(20)
 			playsound(M, 'sound/machines/defib_zap.ogg', 50, 1, -1)
 		return 1
+
+/obj/item/clothing/suit/armor/hockey
+	name = "Ka-Nada winter sport combat suit"
```
Comment (JohnGinnane) [inline code/modules/clothing/suits/armor.dm:421]: Same here, is it "sports" or "sport"?

**Comment by JohnGinnane** [comment]
Code is good I just have minor gripes with capitals and grammar things. English isn't my best so let me know if it's wrong

**Review by JohnGinnane — APPROVED** [review]

**Comment by GuyonBroadway** [comment]
Oh fuck off, one of the other sets just fucked my PR!!

**Comment by GuyonBroadway** [comment]
CLOSING AND REMAKIGN DUE TO SPACEDONG FUCKERY!

**Pull request closed without merging**